packages/um/package.py

```diff
@@ -225,6 +233,44 @@ def check_model_vs_spec(model, config_env, var, spec_value):
                         f"The value {spec_value} will be used.")
 
 
+        def check_model_vs_sources_vs_ref(
+            model,
+            config_env,
+            sources_var,
+            ref_var,
+            resource_path):
+            """
+            Check the values set by the variants sources_var and ref_var
+            against any existing sources value in config_env, and remind
+            that the sources value will be overridden by resource_path.
+            """
+            sources_value = spec.variants[sources_var].value
+            ref_value = spec.variants[ref_var].value
+            tty.info(f"The spec sets {ref_var}={ref_value}")
+            if sources_value == "none":
+                # In this case, the spec has not overridden any value
+                # in the model configuration.
+                if sources_var not in config_env:
+                    tty.info(
+                        f"The {model} model does not specify {sources_var}.")
+                else:  # sources_var in config_env
+                    env_value = config_env[sources_var]
+                    if env_value == "":
+                        tty.info(
+                            f"The {model} model sets {sources_var}=''.")
+                    else:
+                        tty.warn(
+                            f"The {model} model sets "
+                            f"{sources_var}={env_value}.")
+            else:  # sources_value != "none"
+                # In this case, the spec has already overridden any value
+                # in the model configuration.
+                assert sources_value == config_env[sources_var]
```

Comment:
Is this true because of the code in lines 324 to 329?

Reply:
Yes.